src/dashmips.ts

```diff
@@ -25,42 +25,72 @@ export interface DashmipsDebugClient {
 
 export class DashmipsDebugClient extends EventEmitter {
     public dashmipsPid: number = -1
-    private websocket!: WebSocket
+    private socket!: Socket
     private url!: string
+    private cutoff_data: string
+    private cutoff_data_length: number
 
     private _readyNotifier = new Subject()
 
     constructor() {
         super()
+        this.cutoff_data = "";
+        this.cutoff_data_length = 0;
     }
 
-    connect(url: string) {
-        this.url = url
-        this.websocket = new WebSocket(this.url, { handshakeTimeout: 0 })
-        this.websocket.on('open', this.onOpen)
+    connect(host: string, port: number) {
+        this.socket = new Socket()
+        this.socket.on('connect', this.onOpen)
+        this.socket.setEncoding('utf8');
+        this.socket.connect({ port, host })
     }
 
     private onOpen = () => {
-        this.websocket.on('message', this.onMessage)
-        this.websocket.on('close', this.onError)
-        this.websocket.on('error', this.onError)
+        this.socket.on('data', this.onMessage)
+        this.socket.on('close', this.onError)
+        this.socket.on('error', this.onError)
         this._readyNotifier.notify()
     }
 
     private onError = (error?: Error) => {
         this.emit('error', error)
     }
 
-    private onMessage = (message: string) => {
-        const response: DashmipsResponse = JSON.parse(message)
-        if (response.error) {
-            this.emit('error', response.error)
+    private onMessage = (data: string) => {
+        if (this.cutoff_data !== "") {
+            data = "{\"size\": " + String(this.cutoff_data_length) + "}" + this.cutoff_data + data;
```

Comment:
let's clean this up a bit using [template strings](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) and [JSON.stringify](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify):
```js
data = `${JSON.stringify({size: this.cutoff_data_length})}${this.cutoff_data}${data}`
```